module/preference/PreferenceDialog.kt

```diff
@@ -0,0 +1,388 @@
+                        values = values,
+                        selected = selected,
+                        onSelection = { selected = it; state.modified = true }
+                    )
+                }
+            }
+
+            override fun isValid(): Boolean {
+                return true
+            }
+        }
+    }
+
+    class PreferencesForm : State {
+        private val QUERY_LIMIT_PLACEHOLDER = "1000"
+        private val IGNORED_PATHS_PLACEHOLDER = ".git"
+        var modified by mutableStateOf(false)
+
+        // Graph Visualiser Preferences
+        var graphOutput = PreferenceField.Checkbox(
+            this, initialValue = preferenceMgr.graphOutputEnabled, label = ENABLE_GRAPH_OUTPUT,
+            caption = Optional.of(PREFERENCES_GRAPH_OUTPUT_CAPTION)
+        )
+
+        // Project Manager Preferences
+        private val ignoredPathsString = preferenceMgr.ignoredPaths.joinToString(",")
+        var ignoredPaths = PreferenceField.TextInput(
+            this, initialValue = ignoredPathsString,
+            label = PROJECT_IGNORED_PATHS, placeholder = IGNORED_PATHS_PLACEHOLDER
+        )
+
+        // Query Runner Preferences
+        var matchQueryLimit = PreferenceField.TextInput(
+            this,
+            initialValue = preferenceMgr.matchQueryLimit.toString(),
+            label = SET_QUERY_LIMIT, 
+            placeholder = QUERY_LIMIT_PLACEHOLDER,
+            caption = Optional.of(PREFERENCES_MATCH_QUERY_LIMIT_CAPTION)
+        ) {/* validator = */ it.toLongOrNull() != null }
+        
+        // Text Editor Preferences
+        var autoSave = PreferenceField.Checkbox(
+            this, initialValue = preferenceMgr.autoSave, label = ENABLE_EDITOR_AUTOSAVE
+        )
+
+        val preferenceGroups = listOf(
+            PreferenceGroup(GRAPH_VISUALISER, preferences = listOf(graphOutput)),
+            PreferenceGroup(PROJECT_MANAGER, preferences = listOf(ignoredPaths)),
+            PreferenceGroup(QUERY_RUNNER, preferences = listOf(matchQueryLimit)),
+            PreferenceGroup(TEXT_EDITOR, preferences = listOf(autoSave)),
+        )
+
+        val rootPreferenceGroup = PreferenceGroup(entries = preferenceGroups)
+
+        override fun cancel() {
+            StudioState.preference.preferencesDialog.close()
+        }
+
+        fun apply() {
+            if (isValid()) {
+                trySubmit()
+                modified = false
+            }
+        }
+
+        fun ok() {
+            apply()
+            cancel()
+        }
+
+        override fun isValid(): Boolean {
+            return rootPreferenceGroup.isValid()
+        }
+
+
+        override fun trySubmit() {
+            preferenceMgr.autoSave = autoSave.value
+            preferenceMgr.ignoredPaths = ignoredPaths.value.split(',').map { it.trim() }
+            preferenceMgr.matchQueryLimit = matchQueryLimit.value.toLong()
+            preferenceMgr.graphOutputEnabled = graphOutput.value
+        }
```

Comment:
We've kept the logic for submitting the changes to the preferences here as it is the concern of the `PreferencesForm` to save its state rather than an individual `PreferenceField`'s responsibility to save their state via the `PreferenceManager` - a `PreferenceField` concerns the display of a preference rather than the entire concept of a preference within Studio.